src/js/DatePicker.js

```diff
@@ -0,0 +1,436 @@
+        /**
+         * Calendar next month link
+         * @type {JQuery|null}
+         */
+        this.$nextLink = null;
+        /**
+         * Calendar previous month link
+         * @type {JQuery|null}
+         */
+        this.$prevLink = null;
+        /**
+         * Calendar today link
+         * @type {JQuery|null}
+         */
+        this.$todayLink = null;
+        /**
+         * Calendar wrapper
+         * @type {JQuery|null}
+         */
+        this.$datePickerContainer = null;
+
+        this.utils = utils;
+        this.highlightUtils = highlightUtils;
+        this.accessibleLabelUtils = accessibleLabelUtils;
+        this.navigationManager = navigationManager;
+    }
+
+    /**
+     * Initialize
+     * @param {jQuery} $html
+     */
+    init ($html) {
+        this.$html = $html;
+
+        const $datePickers = $html.find('[data-pulsardatepicker="true"]');
+        let defaultDateFormat = 'dd/mm/yy';
+        let inputPlaceholder = 'dd/mm/yyyy';
+
+        $datePickers.each((index, element) => {
+            let $datePickerInput = $(element);
+            let dateFormat = $datePickerInput.attr('data-pulsardatepicker-format');
+            let $linkedTriggerButtonId = $datePickerInput.attr('data-pulsardatepicker-trigger');
+            let $linkedTriggerButton = this.$html.find('#' + $linkedTriggerButtonId);
+
+            if ($datePickerInput.attr('data-pulsardatepicker-trigger') === undefined) {
+                console.warn('Datepicker: The date input must include a data-pulsardatepicker-trigger data attribute with the value matching the ID of the trigger button');
+                return;
+            }
+
+            if (!$linkedTriggerButton.length) {
+                console.warn('Datepicker: The trigger button element with the ID given in data-pulsardatepicker-trigger cannot be found');
+                return;
+            }
+
+            // Check if data-pulsardatepicker-format attribute exists and lowercase it
+            // to eliminate different styles of writing issues
+            if (dateFormat !== undefined) {
+                dateFormat = dateFormat.toLowerCase();
+            }
+
+            // Set date formats
+            switch (dateFormat) {
+                case 'us':
+                    defaultDateFormat = 'mm/dd/yy';
+                    inputPlaceholder = 'mm/dd/yyyy';
+                    break;
+                case 'reverse':
+                    defaultDateFormat = 'yy/mm/dd';
+                    inputPlaceholder = 'yyyy/mm/dd';
+                    break;
+                default:
+                    defaultDateFormat = 'dd/mm/yy';
+                    inputPlaceholder = 'dd/mm/yyyy'
+            }
+
+            // Initialize date pickers
+            $datePickerInput.datepicker({
+                closeText: 'Close',
+                dayNamesShort: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"],
+                dateFormat: defaultDateFormat,
```

Comment:
Probably one for a new ticket, but the datepicker really needs to be posting back it's data in the ISO format used for the `type="date"` input (or have the option to enable it) - which is `yyyy-mm-dd`. I was going to reopen https://github.com/jadu/pulsar/pull/1004 as the date picker doesn't work with the Symfony forms implementation - https://github.com/jadu/pulsar-symfony. We also have a shim in XFP to transform datepickers to submit in the ISO format.

Another idea might even be to allow multiple inputs alongside the date picker for day/month/year like in https://design-system.service.gov.uk/components/date-input/ but with a date picker being show as well

Reply:
Looking into the docs for jquery datepicker - this could probably be achieved a lot more easily than in my orginal PR by utilising `altField` https://jqueryui.com/datepicker/#alt-field

Reply:
Could you add these comments to an issue? Happy to take a look (perhaps next week).